File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        unordered_set<int> seen;
        int count = 1, n = nums.size();

        seen.insert(nums.begin(), nums.end());

        if (seen.size() == n) {
            return 0;
        }

        for (int i = 3; i < n; i += 3) {
            seen.clear();
            bool isDistinct = true;

            for (int j = i; j < n; ++j) {
                if (seen.count(nums[j])) {
                    isDistinct = false;
                    break;
                }
                seen.insert(nums[j]);
            }
            if (isDistinct) {
                return count;
            }
            count++;
        }

        return count;
    }
};

File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        def checkdistinct(arr):
            counter = Counter(arr)

            for val in counter.values():
                if val > 1:
                    return False

            return True

        count = 0
        n = len(nums)

        if checkdistinct(nums):
            return count
        if n < 3:
            return 1

        count += 1

        for i in range(3, n, 3):
            if checkdistinct(nums[i:n]):
                return count
            else:
                count += 1

        return count
